<?xml version="1.0" encoding="UTF-8"?>
<pd:ProcessDefinition xmlns:xsl="http://www.w3.org/1999/XSL/Transform" xmlns:ns="http://www.tibco.com/pe/EngineTypes" xmlns:wsdl="http://schemas.xmlsoap.org/wsdl/" xmlns:wsdlPfx="http://www.tibco.com/CommonLE2/namespace/SecurityManagement/GUIServices" xmlns:ns5="http://www.tibco.com/CommonLE2/namespace/Public/Pluggable/GUI/Configuration/Query" xmlns:ns12="http://www.tibco.com/CommonLE/namespace/Common" xmlns:ns6="http://www.tibco.com/CommonLE2/namespace/Public/Pluggable/GUI/Configuration/Applications" xmlns:ns7="http://www.tibco.com/CommonLE2/namespace/Public/Pluggable/GUI/SecurityConfiguration/UserRolesRequest" xmlns:pfx="http://www.tibco.com/CommonLE2/namespace/Public/Pluggable/GUI/Configuration/RolePermissionInformationRequest" xmlns:pd="http://xmlns.tibco.com/bw/process/2003" xmlns:ns2="http://www.tibco.com/CommonLE2/namespace/Administration/GUIServices/Category" xmlns:ns1="http://www.tibco.com/pe/GenerateErrorActivity/InputSchema" xmlns:ns4="http://www.tibco.com/CommonLE2/namespace/Public/Pluggable/GUI/Configuration/UserRolesPerAppPermissionsRequest" xmlns:ns3="http://www.tibco.com/CommonLE2/Public/Common" xmlns:pfx5="http://www.tibco.com/CommonLE2/namespace/Public/Pluggable/GUI/SecurityConfiguration/AllRolesRetrievalResponse" xmlns:pfx4="http://www.tibco.com/CommonLE2/namespace/Public/Pluggable/GUI/Configuration/UserRolesRequest" xmlns:pfx3="http://www.tibco.com/pe/DeployedVarsType" xmlns:pfx2="http://www.tibco.com/CommonLE2/namespace/private/Common" xmlns:pfx6="http://www.tibco.com/CommonLE2/namespace/Public/Pluggable/GUI/SecurityConfiguration/RolePermissionInformation">
    <xs:import xmlns:xs="http://www.w3.org/2001/XMLSchema" namespace="http://www.tibco.com/CommonLE2/namespace/Public/Pluggable/GUI/SecurityConfiguration/RolePermissionInformation" schemaLocation="/SharedResources/SchemaDefinitions/Public/Pluggable/GUI/Administration/SecurityConfiguration/RolePermissionsInformation.xsd"/>
    <wsdl:import namespace="http://www.tibco.com/CommonLE2/namespace/SecurityManagement/GUIServices" location="/SharedResources/WSDL/GUI/SecurityManagement/intfUpdateRoleACL.wsdl"/>
    <pd:name>SecurityManagement/GUIServices/wsUserRolesApps.process</pd:name>
    <pd:startName>Start</pd:startName>
    <pd:startType wsMsgRef="wsdlPfx:UserRolesRequest"/>
    <pd:startX>63</pd:startX>
    <pd:startY>63</pd:startY>
    <pd:returnBindings>
        <wsdlPfx:Applications>
            <part1>
                <xsl:copy-of select="$Call-UserRolesApps/ns6:Applications"/>
            </part1>
        </wsdlPfx:Applications>
    </pd:returnBindings>
    <pd:endName>End</pd:endName>
    <pd:endType wsMsgRef="wsdlPfx:Applications"/>
    <pd:endX>385</pd:endX>
    <pd:endY>65</pd:endY>
    <pd:errorSchemas>
        <Error wsMsgRef="wsdlPfx:Error"/>
    </pd:errorSchemas>
    <pd:processVariables/>
    <pd:targetNamespace>http://xmlns.example.com/1308688050387</pd:targetNamespace>
    <pd:activity name="Catch Role Retrieval Error">
        <pd:type>com.tibco.pe.core.CatchActivity</pd:type>
        <pd:resourceType>ae.activities.catch</pd:resourceType>
        <pd:x>63</pd:x>
        <pd:y>165</pd:y>
        <pd:handler>true</pd:handler>
        <config>
            <fault>localname=RoleRetrievalError namespace=http://www.tibco.com/CommonLE2/namespace/private/Common</fault>
        </config>
        <pd:inputBindings/>
    </pd:activity>
    <pd:activity name="Call UserRolesApps">
        <pd:type>com.tibco.pe.core.CallProcessActivity</pd:type>
        <pd:resourceType>ae.process.subprocess</pd:resourceType>
        <pd:x>219</pd:x>
        <pd:y>65</pd:y>
        <config>
            <processName>/SecurityManagement/DataAccess/Common/UserRolesApps.process</processName>
        </config>
        <pd:inputBindings>
            <ns7:UserRolesRequest>
                <ns7:UserName>
                    <xsl:value-of select="$Start/wsdlPfx:UserRolesRequest/parameters/ns7:UserRolesRequest/ns7:UserName"/>
                </ns7:UserName>
            </ns7:UserRolesRequest>
        </pd:inputBindings>
    </pd:activity>
    <pd:activity name="Catch All Errors">
        <pd:type>com.tibco.pe.core.CatchActivity</pd:type>
        <pd:resourceType>ae.activities.catch</pd:resourceType>
        <pd:x>64</pd:x>
        <pd:y>335</pd:y>
        <pd:handler>true</pd:handler>
        <config>
            <catchAll>true</catchAll>
        </config>
        <pd:inputBindings/>
    </pd:activity>
    <pd:activity name="Generate Admin Connection Fault">
        <pd:type>com.tibco.pe.core.GenerateErrorActivity</pd:type>
        <pd:resourceType>ae.activities.throw</pd:resourceType>
        <pd:x>219</pd:x>
        <pd:y>165</pd:y>
        <config>
            <faultName>Error</faultName>
        </config>
        <pd:inputBindings>
            <ns1:ActivityInput>
                <message>
                    <xsl:value-of select="$Catch-Role-Retrieval-Error/pfx2:RoleRetrievalError/pfx2:Message"/>
                </message>
                <messageCode>
                    <xsl:value-of select="$Catch-Role-Retrieval-Error/pfx2:RoleRetrievalError/pfx2:Code"/>
                </messageCode>
                <data>
                    <wsdlPfx:Error>
                        <Error>
                            <ns3:Error>
                                <ns3:Message>
                                    <xsl:value-of select="$Catch-Role-Retrieval-Error/pfx2:RoleRetrievalError/pfx2:Message"/>
                                </ns3:Message>
                                <ns3:Code>
                                    <xsl:value-of select="$Catch-Role-Retrieval-Error/pfx2:RoleRetrievalError/pfx2:Code"/>
                                </ns3:Code>
                            </ns3:Error>
                        </Error>
                    </wsdlPfx:Error>
                </data>
            </ns1:ActivityInput>
        </pd:inputBindings>
    </pd:activity>
    <pd:activity name="Generate Internal Error">
        <pd:type>com.tibco.pe.core.GenerateErrorActivity</pd:type>
        <pd:resourceType>ae.activities.throw</pd:resourceType>
        <pd:x>385</pd:x>
        <pd:y>343</pd:y>
        <config>
            <faultName>Error</faultName>
        </config>
        <pd:inputBindings>
            <ns1:ActivityInput>
                <xsl:if test="$_error/ns:ErrorReport/Msg">
                    <message>
                        <xsl:value-of select="$_error/ns:ErrorReport/Msg"/>
                    </message>
                </xsl:if>
                <xsl:if test="$_error/ns:ErrorReport/MsgCode">
                    <messageCode>
                        <xsl:value-of select="$_error/ns:ErrorReport/MsgCode"/>
                    </messageCode>
                </xsl:if>
                <data>
                    <wsdlPfx:Error>
                        <Error>
                            <ns3:Error>
                                <xsl:if test="$_error/ns:ErrorReport/Msg">
                                    <ns3:Message>
                                        <xsl:value-of select="$_error/ns:ErrorReport/Msg"/>
                                    </ns3:Message>
                                </xsl:if>
                                <xsl:if test="$_error/ns:ErrorReport/MsgCode">
                                    <ns3:Code>
                                        <xsl:value-of select="$_error/ns:ErrorReport/MsgCode"/>
                                    </ns3:Code>
                                </xsl:if>
                            </ns3:Error>
                        </Error>
                    </wsdlPfx:Error>
                </data>
            </ns1:ActivityInput>
        </pd:inputBindings>
    </pd:activity>
    <pd:activity name="Local Exception Handler">
        <pd:type>com.tibco.pe.core.CallProcessActivity</pd:type>
        <pd:resourceType>ae.process.subprocess</pd:resourceType>
        <pd:x>219</pd:x>
        <pd:y>343</pd:y>
        <config>
            <processName>/Common/Local Exception Handler.process</processName>
        </config>
        <pd:inputBindings>
            <ns12:Exception>
                <ErrorMessage>
                    <xsl:value-of select="$_error/ns:ErrorReport/Msg"/>
                </ErrorMessage>
                <ErrorCode>
                    <xsl:value-of select="$_error/ns:ErrorReport/MsgCode"/>
                </ErrorCode>
                <ns12:Error>
                    <xsl:for-each xmlns:xsl="http://www.w3.org/1999/XSL/Transform" select="$_error/ns:ErrorReport">
                        <ns12:ErrorReport>
                            <StackTrace>
                                <xsl:value-of select="StackTrace"/>
                            </StackTrace>
                            <Msg>
                                <xsl:value-of select="Msg"/>
                            </Msg>
                            <FullClass>
                                <xsl:value-of select="FullClass"/>
                            </FullClass>
                            <Class>
                                <xsl:value-of select="Class"/>
                            </Class>
                            <ProcessStack>
                                <xsl:value-of select="ProcessStack"/>
                            </ProcessStack>
                            <MsgCode>
                                <xsl:value-of select="MsgCode"/>
                            </MsgCode>
                        </ns12:ErrorReport>
                    </xsl:for-each>
                </ns12:Error>
            </ns12:Exception>
        </pd:inputBindings>
    </pd:activity>
    <pd:activity name="No Records Error">
        <pd:type>com.tibco.pe.core.GenerateErrorActivity</pd:type>
        <pd:resourceType>ae.activities.throw</pd:resourceType>
        <pd:x>219</pd:x>
        <pd:y>244</pd:y>
        <config>
            <faultName>Error</faultName>
        </config>
        <pd:inputBindings>
            <ns1:ActivityInput>
                <message>
                    <xsl:value-of select="$Catch-No-App-Records-Found/pfx2:DBRecordNotFound/pfx2:Message"/>
                </message>
                <messageCode>
                    <xsl:value-of select="$Catch-No-App-Records-Found/pfx2:DBRecordNotFound/pfx2:Code"/>
                </messageCode>
                <data>
                    <wsdlPfx:Error>
                        <Error>
                            <ns3:Error>
                                <ns3:Message>
                                    <xsl:value-of select="$Catch-No-App-Records-Found/pfx2:DBRecordNotFound/pfx2:Message"/>
                                </ns3:Message>
                                <ns3:Code>
                                    <xsl:value-of select="$Catch-No-App-Records-Found/pfx2:DBRecordNotFound/pfx2:Code"/>
                                </ns3:Code>
                            </ns3:Error>
                        </Error>
                    </wsdlPfx:Error>
                </data>
            </ns1:ActivityInput>
        </pd:inputBindings>
    </pd:activity>
    <pd:activity name="Catch No App Records Found">
        <pd:type>com.tibco.pe.core.CatchActivity</pd:type>
        <pd:resourceType>ae.activities.catch</pd:resourceType>
        <pd:x>64</pd:x>
        <pd:y>253</pd:y>
        <pd:handler>true</pd:handler>
        <config>
            <fault>localname=DBRecordNotFound namespace=http://www.tibco.com/CommonLE2/namespace/private/Common</fault>
        </config>
        <pd:inputBindings/>
    </pd:activity>
    <pd:transition>
        <pd:from>Start</pd:from>
        <pd:to>Call UserRolesApps</pd:to>
        <pd:lineType>Default</pd:lineType>
        <pd:lineColor>-16777216</pd:lineColor>
        <pd:conditionType>always</pd:conditionType>
    </pd:transition>
    <pd:transition>
        <pd:from>Catch Role Retrieval Error</pd:from>
        <pd:to>Generate Admin Connection Fault</pd:to>
        <pd:lineType>Default</pd:lineType>
        <pd:lineColor>-16777216</pd:lineColor>
        <pd:conditionType>always</pd:conditionType>
    </pd:transition>
    <pd:transition>
        <pd:from>Catch All Errors</pd:from>
        <pd:to>Local Exception Handler</pd:to>
        <pd:lineType>Default</pd:lineType>
        <pd:lineColor>-16777216</pd:lineColor>
        <pd:conditionType>always</pd:conditionType>
    </pd:transition>
    <pd:transition>
        <pd:from>Local Exception Handler</pd:from>
        <pd:to>Generate Internal Error</pd:to>
        <pd:lineType>Default</pd:lineType>
        <pd:lineColor>-16777216</pd:lineColor>
        <pd:conditionType>always</pd:conditionType>
    </pd:transition>
    <pd:transition>
        <pd:from>Catch No App Records Found</pd:from>
        <pd:to>No Records Error</pd:to>
        <pd:lineType>Default</pd:lineType>
        <pd:lineColor>-16777216</pd:lineColor>
        <pd:conditionType>always</pd:conditionType>
    </pd:transition>
    <pd:transition>
        <pd:from>Call UserRolesApps</pd:from>
        <pd:to>End</pd:to>
        <pd:lineType>Default</pd:lineType>
        <pd:lineColor>-16777216</pd:lineColor>
        <pd:conditionType>always</pd:conditionType>
    </pd:transition>
</pd:ProcessDefinition>